packages/lit-task-auto-top-up/src/actions/getRecipientList.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * `getRecipients`returns singleton instance of the RecipientsList class, which is initialized at
+ * module-scope when this module is loaded.
+ *
+ * We maintain an instance here to facilitate persisting the 'last known good' list of recipients
+ * that we have parsed
+ */
+import getConfig from './getConfig';
+import RecipientList from '../Classes/RecipientList';
+import { RecipientDetail } from '../types/types';
+
+const recipientListInstance = new RecipientList(getConfig());
+
+export default async function getRecipients(): Promise<RecipientDetail[]> {
+  return recipientListInstance.fetchRecipientList();
```

Comment:
Good catch 🤦  Webstorm was dutifully marking the method as unused but for some reason I ignored it 😢  Fixed.

I was torn about this too, because I tend to lean towards 'fail fast and hard' thinking for similar reasons.

My thinking was that if someone was unlikely to see an alert about the list being invalid, then they would miss the alert about the entire service being down, and it would be better to still keep minting for the existing list than to have _everyone_ no longer get tokens... if it takes some time to fix the source, then at least other consumers are being served.

Mainly, what tipped me into implementing this was considering that even if we _do_ have perfect monitoring of the service, _if the file is invalid_, that monitoring won't trigger  and let people who can fix it know_ until the job fails at least once_.  At that point, _we've missed minting for some group of users_, so end-users will be impacted by the error in file format.

This way, we can alert that the source-of-truth is bad without the rest of the previously defined recipients missing a mint.  WDYT?

Also, I'm going to add a commit hook and PR check that validates the JSON matches our schema... but I just know someone will edit it directly in Github and bypass them both with some invalid input sometime.... :P :)

Reply:
Added commit hook and PR checks for the JSON file, and updated the schema to validate the recipient addresses using strict validation provided by `Viem` 🎉 

Given how unlikely it will be for the file to be malformed, given the repository configuration would require that someone _disable branch protection on the repo_, then _use the github web UI to edit `main` directly_ in order to make the file actually end up in `main` in an invalid shape, I think this functionality is now really only about being resilient to transient issues loading the file from `Github`, so I'm going to leave the local cache in for now, unless you have a very strong opinion against it :D